### PR TITLE
Fabo/fetch policy

### LINKED ISSUE #3391: API serves stale balances via GraphQL
Sometimes the ActionModals have stale data, if there was a liveupdate like a tx received.

### PR DESCRIPTION
**Pull request by faboweb**
Closes #3391

instead of setting the fetching policy individually per request this relys on a cache first strategy. the requests go first to the cache so data is there quick. but in the end they still go to the API and if the data changes the components get updated.

https://medium.com/@galen.corey/understanding-apollo-fetch-policies-705b5ad71980

### CODE REVIEW

File: src/ActionModal/components/ActionModal.vue [hunk 1]
```diff
@@ -477,7 +477,6 @@ export default {
         userAddress: this.session.address,
         rewards: this.rewards,
         totalRewards: this.overview.totalRewards,
-        delegations: this.delegations,
```
Comment (jbibla) [inline src/ActionModal/components/ActionModal.vue]: how come this is no longer needed?

File: src/ActionModal/components/ActionModal.vue [hunk 2]
```diff
@@ -731,14 +730,14 @@ export default {
...
             liquidStake
-            totalStake
             accountInformation {
               accountNumber
               sequence
             }
           }
         }
       `,
+      fetchPolicy: "cache-and-network",
```
Comment (jbibla) [inline src/ActionModal/components/ActionModal.vue]: very cool. can we set this globablly or does it have to be set per component?

File: src/components/staking/PageValidator.vue [hunk 3]
```diff
@@ -257,21 +256,8 @@ export default {
...
-    },
-    clearUndelegationCache() {
-      this.$store.commit("invalidateCache", [
-        `overview`,
-        `delegations`,
-        `undelegations`,
-        `transactions`
-      ]) // TODO use more finegrained query string (network and address)
-    },
+    clearDelegationCache() {},
```
Comment (jbibla) [inline src/components/staking/PageValidator.vue]: what does this do?

File: src/vuex/modules/session.js [hunk 4]
```diff
@@ -71,19 +71,6 @@ export default ({ apollo }) => {
     },
     setCurrrentModalOpen(state, modal) {
       state.currrentModalOpen = modal
-    },
-
-    // TODO to own store module?
-    // clear the cache manually so we can force reloading of stale data in not mounted components
-    invalidateCache(state, queries) {
```
Comment (jbibla) [inline src/vuex/modules/session.js]: nice.

**Comment by Bitcoinera** [comment]
It is working really well for Send but not so for Staking. It needs the user to pass by Portfolio in order to show updated available Tokens in the DelegateModal

File: src/ActionModal/components/ActionModal.vue (hunk 2 above)
Comment (faboweb) [reply src/ActionModal/components/ActionModal.vue]: we can, but may want to not do it for things like `networks`. but we could make it the default and set the fetch policy for networks to `cache`

File: src/components/staking/PageValidator.vue (hunk 3 above)
Comment (faboweb) [reply src/components/staking/PageValidator.vue]: nothing, smarta** :P

File: src/ActionModal/components/ActionModal.vue (hunk 1 above)
Comment (faboweb) [reply src/ActionModal/components/ActionModal.vue]: I didn't find any use of delegations in the context. 5 withdraws are calculated based on the rewards.

**Comment by Bitcoinera** [comment]
Still not working

**Comment by faboweb** [comment]
This error does look strange. Why would the validator addresses be `cosmos1` based? I am checking it now

**Comment by faboweb** [comment]
Just staked successfully on https://deploy-preview-3390--lunieio.netlify.com
Could you check again?

**Comment by Bitcoinera** [comment]
Sorry, maybe I didn't explain so clearly (actually at all) what that error earlier with the `cosmosvaloper` was about.

The error happens when you try to delegate to a validator with you available balance in the DelegateModal not yet updated.

I am going to upload a video so you can see better the issue here. I just tested and it is still unresolved.

**Comment by Bitcoinera** [comment]
Now it is all fixed! Updating every time :smile: Also in the last case I commented earlier.

Now there is only one issue left: 

on line 346 in PageValidator, this part (data is undefined):

**Comment by Bitcoinera** [comment]
why don't we do it like so, like it is done in PageProposal and PageProposals?



Like this it also checks that the query has been loaded.

**Comment by faboweb** [comment]
Greate idea

**Pull request merged**